inc/plugins/myalerts.php

```diff
@@ -723,9 +728,12 @@ function myalerts_global_start()
 			$lang->load('myalerts');
 		}
 
-		$mybb->user['myalerts_disabled_alert_types'] = json_decode(
-			$mybb->user['myalerts_disabled_alert_types']
-		);
+		if (!empty($mybb->user['myalerts_disabled_alert_types']))
+		{
+			$mybb->user['myalerts_disabled_alert_types'] = json_decode(
+				$mybb->user['myalerts_disabled_alert_types']
+			);
+		}
```

Comment:
I wonder if it's worth adding an `else` branch here like:

```suggestion
		} else {
				$mybb->user['myalerts_disabled_alert_types'] = array();
		}
```

Reply:
I'd wondered the same thing, and concluded that because the `else` below, on [line 745](https://github.com/MyBBStuff/MyAlerts/blob/e759200270fe250ab04d51ef7853787af42d7736/inc/plugins/myalerts.php#L745), already handles that scenario, the redundancy wasn't worth it. Other than redundancy, though, it seems harmless to me, so, if you prefer to add it, I'll happily commit your suggestion. Just let me know.